src/info/info.ts

```diff
@@ -286,122 +273,200 @@ port.onMessage.addListener(async (message) => {
-                let port = navigator.serial.requestPort();
+            btn.addEventListener("click", async _ => {
+                // const worker = new Worker(browser.runtime.getURL("./info/worker.js"), { type: "module" });
+                let hapticPort = await navigator.serial.requestPort();
+
+                // send all the rendering info
                 worker.postMessage({
-                    renderingData: data,
-                    mode: selectList.value
+                    renderingData: data
                 });
-                //checking for changes in drop down menu
-                selectList.onchange = function () {
-                    worker.postMessage({
-                        renderingData: data,
-                        mode: selectList.value
-                    });
-                };
 
-                worker.addEventListener("message", function (msg) {
 
+                worker.addEventListener("message", function (msg) {
                     // we've selected the COM port
                     btn.style.visibility = 'hidden';
+                   
+
+                    const msgdata = msg.data;
 
                     // return end-effector x/y positions and objectData for updating the canvas
-                    posEE.x = msg.data.positions.x;
-                    posEE.y = msg.data.positions.y;
-                    objectData = msg.data.objectData;
+                    
+                    posEE.x = msgdata.positions.x;
+                    posEE.y = msgdata.positions.y;
+                  
+                    waitForInput = msgdata.waitForInput;
+
+                    // grab segment data if available
+                    if (msgdata.segmentData != undefined)
+                        segments = msgdata.segmentData;
 
-                    // latch to call the refresh of the animation once after which the call is recursive in draw() function
+                    // grab object data if available
+                    if (msgdata.objectData != undefined)
+                        objects = msgdata.objectData;
+
+                    // grab drawing info if available
+                    if (msgdata.drawingInfo != undefined)
+                        drawingInfo = msgdata.drawingInfo;
+
+                    // only request to run draw() once
                     if (firstCall) {
-                        createRect();
-                        window.requestAnimationFrame(draw);
-                        firstCall = false;
+                        if (msgdata.segmentData != undefined ||
+                            msgdata.objectData != undefined) {
+                            window.requestAnimationFrame(draw);
+                            firstCall = false;
+                        }
+                    }
+
+                    // see if the worker wants us to play any audio
+                    if (msgdata.audioInfo != undefined) {
```

Comment:
if (msgData.audioInfo && msgdata.audioInfo.sendAudioSignal ){}
The next if block can be removed